<img src="images/usm.jpg" width="480" height="240" align="left"/>

# MAT281 - Laboratorio N°03

## Objetivos de la clase

* Reforzar los conceptos básicos de pandas.

## Contenidos

* [Problema 01](#p1)


## Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información tal como: edad ,sexo, profesión, etc.

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [2]:
import pandas as pd
import os

In [3]:
# cargar datos
df = pd.read_csv(os.path.join("data","ocupation.csv"), sep="|").set_index('user_id')
df.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

1. ¿Cuál es el número de observaciones en el conjunto de datos?

In [38]:
df.shape[0]

943

Según el comando shape, hay 943 observaciones.

2. ¿Cuál es el número de columnas en el conjunto de datos?

In [5]:
df.shape[1]

4

3. Imprime el nombre de todas las columnas

In [40]:
df.columns.to_frame()

0
age                age
gender          gender
occupation  occupation
zip_code      zip_code

4. Imprima el índice del dataframe

In [39]:
df.index.to_frame()

user_id
user_id         
1              1
2              2
3              3
4              4
5              5
6              6
7              7
8              8
9              9
10            10
11            11
12            12
13            13
14            14
15            15
16            16
17            17
18            18
19            19
20            20
21            21
22            22
23            23
24            24
25            25
26            26
27            27
28            28
29            29
30            30
...          ...
914          914
915          915
916          916
917          917
918          918
919          919
920          920
921          921
922          922
923          923
924          924
925          925
926          926
927          927
928          928
929          929
930          930
931          931
932          932
933          933
934          934
935          935
936          936
937          937
938          938
939          939
940          940
941          941
942          942
943          943

[943 rows x 1 columns]

5. ¿Cuál es el tipo de datos de cada columna?

In [10]:
df.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

Según el dtypes, age son enteros de 64 bits, gender, occupation, zip_code son objetos. Al parecer zip_code
A pesar de estar en formato numérico, son strings.

6. Resumir el conjunto de datos

In [12]:
df.describe()

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

7. Resume conjunto de datos  con todas las columnas

In [33]:
df.describe(include='all')

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

8. Imprimir solo la columna de **occupation**.

In [11]:
df.describe(include='all')[['occupation']]

occupation
count         943
unique         21
top       student
freq          196
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

9. ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?

In [34]:
df.describe(include='all')[['occupation']].loc['unique']

occupation    21
Name: unique, dtype: object

Según la función loc, hay 21 ocupaciones distintas.

10. ¿Cuál es la ocupación más frecuente?

In [35]:
df.describe(include='all')[['occupation']].loc['top']

occupation    student
Name: top, dtype: object

Según loc, la ocupación más frecuente es student.

11. ¿Cuál es la edad media de los usuarios?

In [36]:
df.describe(include='all')[['age']].loc['mean']

age    34.051962
Name: mean, dtype: float64

Según loc, la edad media de los usuarios es de 34 años.

12. ¿Cuál es la edad con menos ocurrencia?

In [37]:
series = df["age"].value_counts()
series_min=series.min()
mask= (series==series_min)
series[mask].to_frame()

age
11    1
10    1
73    1
66    1
7     1

Existen 5 edades con menos ocurrencia, 11,10, 73, 66 y 7 años que hay solo un usuario.